### PR TITLE
opentofu: add new package opentofu v1.6.2

### PR DESCRIPTION
**Pull request by trungams**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

- Add new package opentofu v1.6.2

###### Change Log  

- Add new package opentofu v1.6.2

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Local build
- Buddy build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=542460&view=results

### CODE REVIEW

**Review by amritakohli — APPROVED** [review]

File: SPECS/opentofu/opentofu.spec [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+Release:        3%{?dist}
+Summary:        OpenTofu lets you declaratively manage your cloud infrastructure
+Vendor:         Microsoft Corporation
+Distribution:   Azure Linux
+
+License:        0BSD AND Apache-2.0 AND BSD-2-Clause AND BSD-3-Clause AND MIT AND MPL-2.0
+URL:            https://github.com/opentofu/opentofu
+Source0:        https://github.com/opentofu/opentofu/archive/v%{version}.tar.gz#/%{name}-%{version}.tar.gz
+
+# Below is a manually created tarball, no download link.
```
Comment (mfrw) [inline SPECS/opentofu/opentofu.spec:14]: We have started a convention of adding a shell script that generates the vendor tarball automatically.
I think we might as well add that. 
E.G look at moby-engine/docker-cli etc etc

**Review by mfrw — COMMENTED** [review]
Maybe we should add a shell script to generate vendor tarball

File: SPECS/opentofu/opentofu.spec (hunk 1 above)
Comment (PawelWMS) [reply SPECS/opentofu/opentofu.spec:14]: +1, our automation relies on the `generate_source_tarball.sh` scripts with specific input arguments. There should be plenty of example to use for reference and they all have a very similar structure, so a copy-paste with minor tweaks should do the trick.

File: SPECS/opentofu/opentofu.spec [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+BuildRequires:  golang
+
+%description
+OpenTofu lets you declaratively manage your cloud infrastructure.
+
+%prep
+%autosetup -p1 %{?with_vendor:-b1}
+
+%build
+tar -xf %{SOURCE1} --no-same-owner
```
Comment (PawelWMS) [inline SPECS/opentofu/opentofu.spec:43]: Can we move the untarring into the `%prep` stage where the first source is extracted? You might be able to make use of the `%autosetup (...) -a 1` flag there as well.

File: SPECS/opentofu/opentofu.spec [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+export LDFLAGS="-X github.com/opentofu/opentofu/version.dev=no"
```
Comment (PawelWMS) [inline SPECS/opentofu/opentofu.spec:45]: Why do we need to overwrite our default `LDFLAGS`?  And what's the point of setting this part here and then `-X github.com/opentofu/opentofu/version=$VERSION` separately below?

File: SPECS/opentofu/opentofu.spec [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+install -m 0755 -vp ./bin/*             %{buildroot}%{_bindir}/
+
+%check
+for test in "TestResourceProvider_ApplyCustomWorkingDirectory" \
+            "TestGet_cancel" \
+            "TestInit_cancelModules" \
+            "TestFileExists" \
+; do
+awk -i inplace '/^func.*'"$test"'\(/ { print; print "\tt.Skip(\"disabled failing test\")"; next}1' $(grep -rl $test)
+done
```
Comment (PawelWMS) [inline SPECS/opentofu/opentofu.spec:68]: Why are we disabling these 4 tests? Please add a comment of explanation, so we know what's wrong with them.

File: SPECS/opentofu/opentofu.spec (hunk 1 above)
Comment (trungams) [reply SPECS/opentofu/opentofu.spec]: I added a `generate_source_tarball.sh` for this package.

File: SPECS/opentofu/opentofu.spec (hunk 2 above)
Comment (trungams) [reply SPECS/opentofu/opentofu.spec]: updated

File: SPECS/opentofu/opentofu.spec (hunk 3 above)
Comment (trungams) [reply SPECS/opentofu/opentofu.spec]: Good point. I was mostly following fedora's spec after expanding all their macros. I cleaned up the %build section now.

File: SPECS/opentofu/opentofu.spec (hunk 4 above)
Comment (trungams) [reply SPECS/opentofu/opentofu.spec]: The first 3 come from the original fedora spec file. After checking again I found out they work fine, so they could have failed due to some differences in fedora's build environment. Only the last one, `TestFileExists`, fails for us now. Added a comment in the spec file for that.

**Review by Camelron — APPROVED** [review]

**Pull request merged**